### ISSUE TITLE
Appium cannot launch app in iOS 15 emulator with "Failed to create WDA session" error

### ISSUE THREAD

**Issue by cutePanda123** [appium/appium#16466]
## The problem

After I upgrade XCode to 13.1, Appium cannot launch the iOS app in the iOS 15 emulator with a web driver agent related error.
**Failed to create WDA session (An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: connect ECONNREFUSED 127.0.0.1:8100). Retrying...**

## Environment

* Appium version (or git revision) that exhibits the issue: 1.22.0
* Last Appium version that did not exhibit the issue (if applicable): 1.22.0
* Desktop OS/version used to run Appium: macOS Big Sur 11.6.1 
* Node.js version (unless using Appium.app|exe): 14.17.3
* Npm or Yarn package manager: yarn 1.22.10
* Mobile platform/version under test: iOS 15.0 
* Real device or emulator/simulator: iPhone12ProSimulator
* Appium CLI or Appium.app|exe: appium cli

## Details

After I upgrade XCode to 13.1, appium cannot launch app in the iOS 15 emulator with the below web driver agent related error.
**Failed to create WDA session (An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: connect ECONNREFUSED 127.0.0.1:8100). Retrying...**

I tried a lot of workarounds from other similar issues without being able to make it work. 

For example:
changing wdaStartupRetries, iosInstallPause, wdaStartupRetryInterval from this issue https://github.com/appium/appium/issues/12094#issuecomment-482593020

changing wdaLaunchTimeout
from this issue https://github.com/appium/appium/issues/15976#issuecomment-945065376

## Link to Appium logs
appium log with showXcodeLog and showIOSLog disabled: https://gist.github.com/cutePanda123/fecbb343fc17d847cdaab685c1a0966b
appium log with showXcodeLog and showIOSLog enabled: https://gist.github.com/cutePanda123/0a911282ca33225426c24bf1ab80a045

## Code To Reproduce Issue [ Good To Have ]

N/A

**Comment by mykola-mokhnach** [comment]
Are you sure a correct log is attached? I can observe that showXcodeLog is disabled in both of them.
Also, have you tried to build webdriveragent manually via Xcode?

**Comment by cutePanda123** [comment]
> https://gist.github.com/cutePanda123/0a911282ca33225426c24bf1ab80a045

thanks for the quick response, Mykola! I just updated the showXCodelog with a new link. I have not tried to build web driver agent manually via XCode. I will Google how to do it. If you had one handy, can you please share a wiki/instruction with me?

**Comment by cutePanda123** [comment]
> Are you sure a correct log is attached? I can observe that showXcodeLog is disabled in both of them. Also, have you tried to build webdriveragent manually via Xcode?

thanks for the quick response, Mykola! I just updated the showXCodelog with a new link (https://gist.github.com/cutePanda123/0a911282ca33225426c24bf1ab80a045).  I have not tried to build web driver agent manually via XCode. I will Google how to do it. If you had one handy, can you please share a wiki/instruction with me?

**Comment by mykola-mokhnach** [comment]
You could follow the same [tutorial](https://github.com/appium/appium-xcuitest-driver/blob/master/docs/real-device-config.md#basic-manual-configuration) as for real devices. Just skip the section about configuring digital signatures, they are not needed for Simulator build. Or simply ask your developers to help you

**Comment by mykola-mokhnach** [comment]
> [debug] [35m[WebDriverAgent][39m Beginning test with command 'xcodebuild build-for-testing test-without-building -project /Users/lonli/Codes/BingAdsMobile/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -derivedDataPath /Users/lonli/Library/Developer/Xcode/DerivedData/WebDriverAgent-cjolagzlearxtzgcxjcuqpbxatgo -destination id=0D197F7D-002E-4D57-83EC-469E1F8F3356 IPHONEOS_DEPLOYMENT_TARGET=15.0 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO' in directory '/Users/lonli/Codes/BingAdsMobile/node_modules/appium-webdriveragent'


you could also simply start this command from Terminal and observe the ouput

**Comment by cutePanda123** [comment]
> > [debug] �[35m[WebDriverAgent]�[39m Beginning test with command 'xcodebuild build-for-testing test-without-building -project /Users/lonli/Codes/BingAdsMobile/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -derivedDataPath /Users/lonli/Library/Developer/Xcode/DerivedData/WebDriverAgent-cjolagzlearxtzgcxjcuqpbxatgo -destination id=0D197F7D-002E-4D57-83EC-469E1F8F3356 IPHONEOS_DEPLOYMENT_TARGET=15.0 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO' in directory '/Users/lonli/Codes/BingAdsMobile/node_modules/appium-webdriveragent'
> 
> you could also simply start this command from Terminal and observe the ouput

this is so helpful, thanks for sharing! per the output, XCode successfully builds the web driver agent.
logs are here https://gist.github.com/cutePanda123/55c3498c56103037d235595b9f83feed

I can also saw the webdriveragent app in the emulator:
![image](https://user-images.githubusercontent.com/26092755/154368161-a38661e3-4bf0-4b93-bd77-66bd99ecc975.png)

**Comment by mykola-mokhnach** [comment]
> 2022-02-16 14:25:16.056698-0800 WebDriverAgentRunner-Runner[98724:1076985] -[UIKeyboardImpl setSoftwareKeyboardShownByTouch:]: unrecognized selector sent to instance 0x7fee4a81d000
> <unknown>:0: error: UITestingUITests : -[UIKeyboardImpl setSoftwareKeyboardShownByTouch:]: unrecognized selector sent to instance 0x7fee4a81d000

This exception means that webdriveragent sources are out of date and don't support the given iOS version. Try to delete and  fully reinstall Appium to make sure WDA is up to date. We do not maintain Appium 1 anymore, so consider switching to Appium 2 to get the most recent fixes.

**Comment by KazuCocoa** [comment]
yea, it should be fixed in 1.22.0 as Xcode 13 support. Probably the xcodebuild local cache or something remained.

**Comment by mykola-mokhnach** [comment]
Just to add on Xcode caches cleanup: https://developerinsider.co/clean-xcode-cache-quick-fix/

**Comment by cutePanda123** [comment]
thanks, @KazuCocoa  @mykola-mokhnach . I tried to clean Xcode in my local but the issue is still here. Also, I have a build pipeline that runs appium test in a clean macOS image every time and the issue shows there too. Thus, I do not think the root cause is related to XCode local cache. 

I also tried different versions of appium-webdriveragent (3.9.2, 3.16.0, and 4.1.3). None of them can resolve the issue. 

Anything else I can try?
